pkg/types/validation/installconfig_test.go

```diff
@@ -1413,6 +1413,16 @@ func TestValidateInstallConfig(t *testing.T) {
 			}(),
 			expectedError: `^compute\[0\].architecture: Invalid value: "arm64": heteregeneous multi-arch is not supported; compute pool architecture must match control plane$`,
 		},
+		{
```

Comment:
do we need to add this for Azure and GCP as well?

Reply:
I'll do in the respective PRs adding support in those platforms.